### ISSUE TITLE
Incompatibility with user32 library functions

### ISSUE THREAD

**Issue by jrnepic** [nvaccess/nvda#7336]
### Steps to reproduce:
1. Open a Windows desktop program that uses these user32.dll APIs to create a pop-up menu: CreatePopupMenu, InsertMenuItem, and TrackPopupMenu.

### Expected behavior:
The pop-up menu should open without crashing the program.

### Actual behavior:
The program crashes and NVDA seems to restart itself. If I try debugging, I see an error message: "Unhandled exception at 0x2E150B66 (VBufBackend_mshtml.dll)"

### System configuration:
NVDA version:
NVDA 2017.2 from the nvdaaccess.org website.

NVDA Installed or portable:
Installed.

Other information:
Here's what NVDA logged during and after the error: 
[nvda_error.txt](https://github.com/nvaccess/nvda/files/1109817/nvda_error.txt)

Windows version:
Windows 10 Enterprise Version 1511 Build 10586.962

Name and version of other software in use when reproducing the issue:
I saw this issue while using a VB program that can only be accessed with a license.

### Other questions:

Does the issue still occur after restarting your PC?
Yes.

Have you tried any other versions of NVDA?
I tested with 2017.1 and couldn't reproduce the issue with that version.

**Comment by jcsteh** [comment]
This is definitely not related to these User32 functions, as a huge number of applications use these functions and NVDA works as expected with them. What made you suspect these functions?

You noted "the program crashes". I assume you meant the application you were using? If NVDA restarts, it sounds like NVDA is crashing as well.

This is going to be very tricky to investigate given that we cannot access the application. In order to debug this, we're going to need crash dumps from both the application and NVDA. See [Log Files and Crash Dumps](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps) for details on how to obtain these.

**Comment by jrnepic** [comment]
I suspect these functions because I observed that the crash occurs at TrackPopupMenu. Breakpoints before that function call are always hit. Breakpoints after that call are not hit if there is a crash. Note that this crash doesn't happen every time, so there must also be other contributing factors. I tried testing with an empty menu and noticed that the issue goes away. Adding anything to the menu with InsertMenuItem will cause the crash to occur. I'll try to get some crash dumps to make debugging easier.
